### PR TITLE
Add multi-cluster kind setup for testing

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

This is the initial setup script for running e2e tests across multiple kind clusters.

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

## Discussion

*Are there any design details that you would like to discuss further?*

## Testing

Locally created the setup: 

```bash
$ kind get clusters
cluster1
cluster2
cluster3
cluster4
```

Tested with the multi-dc setup:

```bash
$ kubectl --kubeconfig $KUBECONFIG1 exec -ti test-cluster-dc1-storage-3  -- bash
[root@test-cluster-dc1-storage-3 /]# fdbcli
Using cluster file `/var/dynamic-conf/fdb.cluster'.

The database is available.

Welcome to the fdbcli. For help, type `help'.
fdb> status

Using cluster file `/var/dynamic-conf/fdb.cluster'.

Configuration:
  Redundancy mode        - double
  Storage engine         - ssd-2
  Coordinators           - 9
  Desired Commit Proxies - 2
  Desired GRV Proxies    - 1
  Desired Resolvers      - 1
  Desired Logs           - 3
  Desired Remote Logs    - 3
  Desired Log Routers    - 3
  Usable Regions         - 2
  Regions:
    Primary -
        Datacenter                    - dc1
        Satellite datacenters         - dc3, dc2
        Satellite Logs                - 3
    Remote -
        Datacenter                    - dc2
        Satellite datacenters         - dc3, dc1
        Satellite Logs                - 3

Cluster:
  FoundationDB processes - 10
  Zones                  - 10
  Machines               - 10
  Memory availability    - 8.0 GB per process on machine with least available
  Fault Tolerance        - 1 machines
  Server time            - 11/21/22 14:10:01
```

I will do some more testing and bring up an FDB cluster in that setup.

## Documentation

I will add the required documentation before merging.

## Follow-up

Create the e2e tests :)

### CODE REVIEW

**Review by johscheuer — COMMENTED** [review]
Thea idea is to convert those tests: https://github.com/FoundationDB/fdb-kubernetes-operator/tree/main/config/tests/multi_dc to use the new kind setup.

File: config/tests/multi_dc/create.bash [hunk 1]
```diff
@@ -15,14 +17,24 @@ DIR="${BASH_SOURCE%/*}"
 
 . $DIR/functions.bash
 
-applyFile $DIR/stage_1.yaml dc1 '""'
-checkReconciliationLoop test-cluster-dc1
-connectionString=$(getConnectionString test-cluster-dc1)
+# To test a multi-region FDB cluster setup we need to have 4 Kubernetes clusters
```
Comment (johscheuer) [inline config/tests/multi_dc/create.bash:20]:
```suggestion
# To test a multi-region FDB cluster setup we need to have 3 Kubernetes clusters
```

File: config/tests/multi_dc/delete.bash [hunk 2]
```diff
@@ -1,3 +1,17 @@
-#! /bin/bash
+#!/usr/bin/env bash
 
-kubectl delete fdb -l cluster-group=test-cluster
+set -eu
+
+# To test a multi-region FDB cluster setup we need to have 4 Kubernetes clusters
```
Comment (johscheuer) [inline config/tests/multi_dc/delete.bash:5]:
```suggestion
# To test a multi-region FDB cluster setup we need to have 3 Kubernetes clusters
```

**Comment by sbodagala** [comment]
I took a look, but probably this needs to be reviewed by people who know about kind and have a much deeper understanding of the workings of Kubernetes.

**Review by sbodagala — APPROVED** [review]

**Pull request merged**